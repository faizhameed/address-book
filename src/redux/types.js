export const actionTypes = {
  ADD_BOOK: "ADD_BOOK",
};
