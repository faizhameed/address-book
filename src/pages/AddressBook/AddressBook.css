/* No CSS *//*# sourceMappingURL=AddressBook.css.map */