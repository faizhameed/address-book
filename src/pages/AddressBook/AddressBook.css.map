{
    "version": 3,
    "mappings": "",
    "sources": [
        "AddressBook.scss"
    ],
    "names": [],
    "file": "AddressBook.css"
}