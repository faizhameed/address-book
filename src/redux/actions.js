import { actionTypes } from "./types";
import Axios from "axios";

export const fetchContactList = (page = 1) => {
  let url = `https://randomuser.me/api/?${page}&results=50`;

  return function (dispatch) {
    dispatch({
      type: actionTypes.GET_CONTACTS_PENDING,
    });
    Axios.get(url)
      .then((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_SUCCESS,
          payload: response.data,
        })
      )
      .catch((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_FAILED,
          payload: response.error,
        })
      );
  };
};

export const updatePageNumber = (page) => ({
  type: actionTypes.UPDATE_PAGE_NUMBER,
  payload: page,
});

export const searchFieldChange = (data) => ({
  type: actionTypes.SEARCH_FIELD_CHANGE,
  payload: data,
});

export const updatePersonDetailView = (person) => ({
  type: actionTypes.UPDATE_PERSON_DETAIL,
  payload: person,
});

export const toggleDarkMode = () => ({
  type: actionTypes.TOGGLE_DARK_MODE,
});
