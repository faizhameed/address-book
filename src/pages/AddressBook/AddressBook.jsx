import React, { useEffect, lazy, Suspense } from "react";
import { connect } from "react-redux";
import { fetchContactList } from "../../redux/actions";
import Spinner from "../../components/spinner/spinner.component";
import Header from "../../components/Header/Header";
import SearchBar from "../../components/SearchBar/SearchBar";
/**
 * code spliting contactTable so that spinner is enabled to improve loading and UX
 */
const ContactTable = lazy(() => import("../../components/table/contactTable"));

const AddressBook = ({ fetchContactList, pageNumber }) => {
  useEffect(() => {
    /**
     * fetch data with page number
     * whenever page number is updated fetch is run automatically
     * we run useEffect for every pagenumber updation
     */
    fetchContactList(pageNumber);
  }, [fetchContactList, pageNumber]);
  return (
    <React.Fragment>
      <Header content={`Address Book`} />
      <div className="main">
        <Suspense fallback={<Spinner />}>
          <SearchBar />
          <h4>New data will not be fetched if search field has values</h4>
          <h3>Click on name to view details</h3>
          <ContactTable />
        </Suspense>
      </div>
    </React.Fragment>
  );
};

const mapStateToProps = ({ contactListReducer: { page } }) => ({
  pageNumber: page,
});

const mapDispatchToProps = (dispatch) => ({
  fetchContactList: () => dispatch(fetchContactList()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AddressBook);
