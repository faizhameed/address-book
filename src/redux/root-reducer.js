import { combineReducers } from "redux";
import { actionTypes } from "./types";

const initialStateContactList = {
  contactList: [],
  error: null,
  isPending: false,
  page: 1,
};
function contactListReducer(state = initialStateContactList, action) {
  switch (action.type) {
    case actionTypes.GET_CONTACTS_SUCCESS:
      return {
        ...state,
        contactList: [...state.contactList, ...action.payload.results],
        isPending: false,
      };

    case actionTypes.GET_CONTACTS_PENDING:
      return {
        ...state,
        isPending: true,
      };
    case actionTypes.GET_CONTACTS_FAILED:
      return {
        ...state,
        isPending: false,
        error: action.payload,
      };

    case actionTypes.UPDATE_PAGE_NUMBER:
      if (action.payload < 11) {
        return {
          ...state,
          page: action.payload,
        };
      } else {
        return state;
      }
    default:
      break;
  }
  return state;
}

const rootReducer = combineReducers({
  contactListReducer,
});

export default rootReducer;
