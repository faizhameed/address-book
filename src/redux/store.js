import { createStore } from "redux";
import rootReducer from "./root-reducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const logger = createLogger();
export const store = createStore(
  rootReducer,
  applyMiddleware(thunkMiddleware, logger)
);
