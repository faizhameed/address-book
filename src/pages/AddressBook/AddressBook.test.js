import { shallow } from "enzyme";
import AddressBook from "./AddressBook";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("Address Book  ", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <AddressBook />
      </Provider>
    );
  });

  it("Address Book Page is rendered Properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
