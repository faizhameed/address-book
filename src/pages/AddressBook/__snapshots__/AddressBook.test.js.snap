// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Book   Address Book Page is rendered Properly 1`] = `ShallowWrapper {}`;
